src/transformers/models/maskformer/modeling_maskformer.py

```diff
@@ -0,0 +1,2568 @@
+# coding=utf-8
+# Copyright 2022 Facebook AI Research and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch MaskFormer model."""
+
+import collections.abc
+import math
+import random
+from dataclasses import dataclass
+from numbers import Number
+from typing import Dict, List, Optional, Tuple
+
+import numpy as np
+import torch
+import torch.distributed as dist
+from torch import Tensor, nn
+from torch.nn.functional import binary_cross_entropy_with_logits, cross_entropy, interpolate
```

Comment:
We can go for classes, but I'll have to convert all the losses to classes as well

Reply:
I don't see why you can't use the classes inside your functions.

Reply:
Yes for consistency, please use the classes as in all other models.

Reply:
Apologies, but I don't understand. Should I instantiate `binary_cross_entropy_with_logits` inside my loss function or should I create a class for my loss function with an instance of `binary_cross_entropy_with_logits` in its constructor?

Reply:
No you can just import `BCEWithLogitsLoss`

Reply:
sure, I was wondering if we want to avoid creating and throwing away a class instance every time we call the function since `BCEWithLogitsLoss` internally creates a couple of buffers. I am not an expert on this side, so take this comment with a grain of salt

Reply:
As NielsRogge said, this is what is done in all other modeling files. Loss module don't have parameters, so I doubt anything created internally takes a lot of time/memory space.

Reply:
Okay, thanks